### PR TITLE
Improvement / add api calls to round robin view

### PR DESCRIPTION
**Pull request by pinskii**

### CODE REVIEW

**Comment by melinaruu** [comment]
still to do is updating  the table based on match results

File: frontend/src/components/modules/Tournaments/OngoingTournament/RoundRobinTournamentView.tsx [hunk 1]
```diff
@@ -12,64 +13,206 @@ import {
...
+          match.elapsedTime > 0 &&
+          match.winner !== undefined
+      );
+    }
+  };
+
+  const createMatchButtons = (): void => {
+    const createMatchButton = (
+      match: Match,
+      variant: "contained" | "contained" | "contained",
```
Comment (vilbedawg) [inline frontend/src/components/modules/Tournaments/OngoingTournament/RoundRobinTournamentView.tsx:115]: "contained" is passed twice here

File: frontend/src/components/modules/Tournaments/OngoingTournament/RoundRobinTournamentView.tsx [hunk 2]
```diff
@@ -12,64 +13,206 @@ import {
...
-  }, []);
+    const ongoingElements = ongoingMatches.map((match, index) =>
+      createMatchButton(match, "contained")
+    );
+    const upcomingElements = upcomingMatches.map((match, index) =>
+      createMatchButton(match, "contained", true)
+    );
+    const pastElements = pastMatches.map((match, index) =>
+      createMatchButton(match, "contained", false, "secondary")
+    );
```
Comment (vilbedawg) [inline frontend/src/components/modules/Tournaments/OngoingTournament/RoundRobinTournamentView.tsx:150]: I think that you could change the function signature of the `createMatchButton` to accept an object of type `ButtonProps`:

```tsx
const createMatchButton = (
  match: Match,
  props: ButtonProps
): JSX.Element => {
  ...
  return (
    <div style={{ marginBottom: "10px" }}>
      <Button
        onClick={() => {
          navigate(`/matches/${match.id}`);
        }}
        {...props}
      >
      ...
      </Button>
    </div>
  );
};
```

You could then pass any additional props like so:

```tsx
const pastElements = pastMatches.map((match, index) =>
  createMatchButton(match, {
    disabled: false,
    variant: "contained",
    color: "secondary"
  })
);
```

File: frontend/src/components/modules/Tournaments/OngoingTournament/RoundRobinTournamentView.tsx [hunk 3]
```diff
@@ -12,64 +13,206 @@ import {
...
+            }
+          }
+          processedMatches.add(match.id);
+        }
+      }
+      return updatedPlayers;
+    });
+  };
+
+  const generateTableCells = (player: tournamentPlayer): ReactNode[] => {
```
Comment (vilbedawg) [inline frontend/src/components/modules/Tournaments/OngoingTournament/RoundRobinTournamentView.tsx:196]: Use an object iterator to go through the player properties,  i.e., 

```tsx
const generateTableCells = (player: tournamentPlayer): ReactNode[] => {
  return Object.values(player).map((value, index) => {
    if (index === 0) {
      // If we want to skip the ID property
      return null;
    }

    return (
      <TableCell key={index}>
        <Typography>{value}</Typography>
      </TableCell>
    );
  });
};

File: frontend/src/components/modules/Tournaments/OngoingTournament/RoundRobinTournamentView.tsx [hunk 4]
```diff
@@ -12,64 +13,206 @@ import {
...
 } from "@mui/material";
 import "react-tabs/style/react-tabs.css";
+import { type User, type Match } from "types/models";
+import { useNavigate } from "react-router-dom";
+import { useTournament } from "context/TournamentContext";
 
-interface NamesState {
-  player1: string;
-  player2: string;
+interface tournamentPlayer {
```
Comment (vilbedawg) [inline frontend/src/components/modules/Tournaments/OngoingTournament/RoundRobinTournamentView.tsx:20]: interfaces and class names should be PascalCased. `TournamentPlayer`

**Review by vilbedawg — APPROVED** [review]
Other than the few minor suggestions, looks good to me! Awesome work!

**Comment by shepherd-06** [comment]
There was a build error kept happening in upcoming tournament view. So I added player details. 

image

**Comment by vilbedawg** [comment]
> There was a build error kept happening in upcoming tournament view....

There was an fix for it already but oh well :D 
https://github.com/Koodikukkaro/Kendo-tournament-app/pull/64/files

**Pull request merged**